src/ipc/ipc4/handler.c

```diff
@@ -744,7 +744,7 @@ static int ipc4_module_process_dx(union ipc4_message_header *ipc4)
 	struct ipc4_module_set_dx dx;
 	struct ipc4_dx_state_info dx_info;
 	uint32_t module_id, instance_id;
-	uint32_t core_id;
+	int32_t core_id;
```

Comment:
Would a better fix be to only loop for ```core_id < CONFIG_CORES``` ?

Reply:
And primary core would be handled outside of the loop? I don't know if it is clearer.